pkg/plugins/policies/meshmetric/plugin/xds/prometheus_configurer.go

```diff
@@ -0,0 +1,130 @@
+package xds
+
+import (
+	"github.com/kumahq/kuma/pkg/core"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	core_xds "github.com/kumahq/kuma/pkg/core/xds"
+	api "github.com/kumahq/kuma/pkg/plugins/policies/meshmetric/api/v1alpha1"
+	envoy_common "github.com/kumahq/kuma/pkg/xds/envoy"
+	envoy_clusters "github.com/kumahq/kuma/pkg/xds/envoy/clusters"
+	envoy_listeners "github.com/kumahq/kuma/pkg/xds/envoy/listeners"
+)
+
+var log = core.Log.WithName("MeshMetric")
+
+type PrometheusConfigurer struct {
+	Backend         *api.PrometheusBackend
+	ClusterName     string
+	ListenerName    string
+	EndpointAddress string
+	StatsPath       string
+}
+
+func (pc *PrometheusConfigurer) ConfigureCluster(proxy *core_xds.Proxy) (envoy_common.NamedResource, error) {
+	return envoy_clusters.NewClusterBuilder(proxy.APIVersion, pc.ClusterName).
+		Configure(envoy_clusters.ProvidedEndpointCluster(proxy.Dataplane.IsIPv6(),
+			core_xds.Endpoint{
+				UnixDomainPath: proxy.Metadata.MetricsSocketPath,
+			},
+		)).
+		Configure(envoy_clusters.DefaultTimeout()).
+		Build()
+}
+
+func (pc *PrometheusConfigurer) ConfigureListener(proxy *core_xds.Proxy, mesh *core_mesh.MeshResource) (envoy_common.NamedResource, error) {
+	var listener envoy_common.NamedResource
+	var err error
+
+	// nolint:gocritic
+	if pc.useMeshMtls(mesh) {
+		listener, err = pc.mtlsListener(proxy, mesh)
+		if err != nil {
+			return nil, err
+		}
+	} else if pc.useProvidedTls(proxy.Metadata) {
+		listener, err = pc.providedTlsListener(proxy)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		listener, err = pc.unsecuredListener(proxy)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return listener, nil
+}
+
+func (pc *PrometheusConfigurer) useMeshMtls(mesh *core_mesh.MeshResource) bool {
+	return pc.Backend.Tls != nil && pc.Backend.Tls.Mode == api.ActiveMTLSBackend && mesh.MTLSEnabled()
```

Comment:
I think we talked about not having `ActiveMTLSBackend`, right?

Reply:
yes, I didn't have time to remove it, now it is gone